test/extended/util/framework.go

```diff
@@ -2106,3 +2107,37 @@ func DoesApiResourceExist(oc *CLI, apiResourceName string) (bool, error) {
 
 	return false, nil
 }
+
+var (
+	microShiftVersion string
+	microShiftMutex   sync.Mutex
+)
+
+func getMicroShiftVersion(ocClient *CLI) (string, error) {
+	microShiftMutex.Lock()
+	defer microShiftMutex.Unlock()
+
+	if microShiftVersion == "" {
```

Comment:
This function would be simpler if this test was reversed and it returned the cached value immediately.

Reply:
I thought of doing the same, but I chose to follow the existing pattern here: https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L2002-L2012

There is a kind of early return but its only under some special conditions a few lines below: https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L2030-L2042

Should I keep the style?

Reply:
ok, let's keep this version